src/ng/directive/form.js

```diff
@@ -514,6 +514,13 @@ var formDirectiveFactory = function(isNgForm) {
     };
 
     return formDirective;
+
+    function getSetter(expression) {
+      if (expression === '') {
+        return $parse('this[""]').assign;
```

Comment:
410f7c682633c681be641cd2a321f9e51671d474 added some code changes and a test so when the name is the empty string, then the form object is stored in the scope under the `''` name. This code workarounds this case as the empty expression is not assignable